### PR TITLE
Yermishyn (IM-32) winter-2024-exam

### PR DESCRIPTION
**Pull request by ErmishinS**

### CODE REVIEW

File: Tasks/between.js [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
+  const startIndex = str.indexOf(prefix);
+
+  if (startIndex === -1) {
+    return '';
   }
-  return str;
+
+  const substring = str.substring(startIndex + prefix.length);
+
+  return suffix ? substring.substring(0, substring.indexOf(suffix)) : substring;
```
Comment (tshemsedinov) [inline Tasks/between.js:16]: It's better to put position into intermediate identifier as well as in prev case

File: Tasks/distinct.js [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+// distinct.js is refactored, Test distinct.js: Passed: 9 of 9
+
+'use strict';
+
+const distinct = (arr) => {
+  const uniqueSet = new Set();
+  
+  for (const element of arr) {
+    uniqueSet.add(element);
+  }
```
Comment (tshemsedinov) [inline Tasks/distinct.js:12]: Set can be populated without loop, just with constructor call

File: Tasks/merge.js [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
-  object_3 = [{}][0];
-  for (attribute_name in object_1) {
-    object_3[attribute_name] == object_1[attribute_name];
-    object_3[attribute_name] = object_1[attribute_name];
+// merge.js is refactored, Test merge.js: Passed: 6 of 6
+
+'use strict';
+
+const mergeTwoObjects = (object1, object2) => {
+  const merged_dict = {};
```
Comment (tshemsedinov) [inline Tasks/merge.js:8]: Naming conventions....

File: Tasks/between.js (hunk 1 above)
Comment (ErmishinS) [reply Tasks/between.js:17]: Fixed

File: Tasks/distinct.js (hunk 2 above)
Comment (ErmishinS) [reply Tasks/distinct.js]: Fixed

File: Tasks/merge.js (hunk 3 above)
Comment (ErmishinS) [reply Tasks/merge.js]: Fixed

**Pull request closed without merging**